plugins/module_utils/k8s/waiter.py

```diff
@@ -66,7 +66,7 @@ def statefulset_ready(statefulset: ResourceInstance) -> bool:
     ready_replicas = statefulset.status.readyReplicas or 0
     return bool(
         statefulset.status
-        and statefulset.spec.updateStrategy.type == "RollingUpdate"
+        and statefulset.spec.updateStrategy.type in ["RollingUpdate", "OnDelete"]
```

Comment:
This one is a bit tricky. The problem with StatefulSets that use OnDelete is that the currentRevision and updateRevision may not match, since pods aren't replaced until they are manually deleted. In some cases, updatedReplicas may also not match the spec's replicas. You can see this by creating a StatefulSet using the OnDelete update strategy and more than one replica. Then, in a separate task, modify the container spec. Now, scale it down to 1 replica.

I think what we need here is two sets of conditions, one for RollingUpdate and one for OnDelete. The block for OnDelete should not check the revision or the updated replicas. So something like:
```python
if statefulset.spec.updateStrategy.type == "OnDelete":
    return bool(...) # modified condition checks
return bool(...) # current condition checks
```